Automattic/wp-calypso
# Refactor types in `calypso-products`

### Changes proposed in this Pull Request

This PR refactors types of the `calypso-products` package, mainly to avoid duplicates.

### Implementation notes

- Added the `const` type in `packages/calypso-products/src/constants/jetpack.ts`
- Renamed `JETPACK_PRODUCTS_LIST_WITH_FEATURES` to `JETPACK_SITE_PRODUCTS_WITH_FEATURES`, to avoid confusion
- Merged content of `packages/calypso-products/src/product-values-types.ts`
 to `packages/calypso-products/src/types.ts`

### Testing instructions

- Review code
- Check that all tests pass
- Make sure you can boot up Calypso and cloud successfully
